python/spacewalk/satellite_tools/repo_plugins/yum_src.py

```diff
@@ -536,11 +536,19 @@ def _get_mirror_list(self, repo, url):
         verify = self.sslcacert
         try:
             webpage = requests.get(url, proxies=proxies, cert=cert, verify=verify)
-            content_type = webpage.headers["Content-Type"]
-            # amazonlinux core channels content-type = binary/octet-stream
-            if "text/plain" not in content_type and "xml" not in content_type and "octet-stream" not in content_type:
-                # Not a valid mirrorlist or metalink; continue without it
-                return returnlist
+            # We want to check the page content-type usually, but
+            # we have to wrap the next bit in a try-block for if the resource is
+            # cached and returns a 304; cached page returns no content type
+            # (if page is cached, for now we will assume it is the right type)
+            try:
+                content_type = webpage.headers["Content-Type"]
```

Comment:
I wouldn't use `try..except` here, but instead simply something like: 

```python
content_type = webpage.headers.get("Content-Type")
if content_type and "text/plain" not in content_type and "xml" not in content_type and "octet-stream" not in content_type:
    return returnlist
elif not content_type:
    # This will then go straight to the next try block.
    log(1, "No content-type header. Treating as valid.")

Reply:
I'd say this comes down to if we can expect a "Content-Type" key or not. If it's there 90% (just to give a number) of the time we check I'd go with `try: ... except KeyError:`, otherwise with `.get()` . 

The reason is that, if we decide which one to use (both work) base on which outcome is more common, we can then use that information in the future to understand the code a bit easier.

Reply:
Okay, there should usually be a key so I'll keep the try block and just make it specifically check for a KeyError.